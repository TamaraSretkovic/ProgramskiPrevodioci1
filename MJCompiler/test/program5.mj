//	Semantika test 
//  - nedefinisan tip bla

program test302
	int niz[];
{
	void main() {
		int = bla;
	}
}
