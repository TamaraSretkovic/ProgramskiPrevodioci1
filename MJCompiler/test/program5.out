INFO  01:15:54,452 - Compiling source file: /home/tamarica/Documents/tamara/velika skola/8 semestar/PP1/projekat/projecat_2019_2020/MJCompiler/test/program5.mj
INFO  01:15:54,476 - #2 program
INFO  01:15:54,476 - #46 test302
INFO  01:15:54,476 - #46 int
INFO  01:15:54,476 - #46 niz
INFO  01:15:54,477 - #8 [
INFO  01:15:54,477 - #9 ]
INFO  01:15:54,477 - #5 ;
INFO  01:15:54,477 - #3 {
INFO  01:15:54,478 - #10 void
INFO  01:15:54,478 - #46 main
INFO  01:15:54,478 - #12 (
INFO  01:15:54,478 - #11 )
INFO  01:15:54,479 - #3 {
INFO  01:15:54,479 - #46 int
INFO  01:15:54,479 - #23 =
INFO  01:15:54,479 - #46 bla
INFO  01:15:54,480 - #5 ;
INFO  01:15:54,480 - #4 }
INFO  01:15:54,480 - #4 }
INFO  01:15:54,485 - ==================SINTAKSNO STABLO==================
INFO  01:15:54,485 - Program(
  ProgName(
   test302
  ) [ProgName]
  DeclarationListDeclaration(
    NoDeclarationList(
    ) [NoDeclarationList]
    DeclarationListVar(
      VarDeclarationV(
        VarType(
          Type(
           int
          ) [Type]
        ) [VarType]
        VarDecl(
          IdentArray(
           niz
          ) [IdentArray]
        ) [VarDecl]
      ) [VarDeclarationV]
    ) [DeclarationListVar]
  ) [DeclarationListDeclaration]
  MethodDeclarationListMethod(
    NoMethodDeclarationList(
    ) [NoMethodDeclarationList]
    VoidMethodDeclaration(
      MethodVoidName(
       main
      ) [MethodVoidName]
      MethodDeclarationActual(
        NoFormalParamsList(
        ) [NoFormalParamsList]
        NoDeclarationList(
        ) [NoDeclarationList]
        StatementListStatement(
          NoStatementList(
          ) [NoStatementList]
          DesignatorStatementStatement(
            LeftDesignatorAssignExpression(
              LeftSideAssignVar(
                DesignatorOnly(
                 int
                ) [DesignatorOnly]
              ) [LeftSideAssignVar]
              AssignOpEqual(
                AssignOp(
                ) [AssignOp]
              ) [AssignOpEqual]
              ExpressionTermDef(
                OnlyTerm(
                  TermFactor(
                    FactorDesignator(
                      DesignatorOnly(
                       bla
                      ) [DesignatorOnly]
                    ) [FactorDesignator]
                  ) [TermFactor]
                ) [OnlyTerm]
              ) [ExpressionTermDef]
            ) [LeftDesignatorAssignExpression]
          ) [DesignatorStatementStatement]
        ) [StatementListStatement]
      ) [MethodDeclarationActual]
    ) [VoidMethodDeclaration]
  ) [MethodDeclarationListMethod]
) [Program]
INFO  01:15:54,485 - ==================SEMANTICKA OBRADA==================
INFO  01:15:54,488 - Deklarisan niz niz(Var niz: array, 0, 0 ) na liniji 5
INFO  01:15:54,488 - Obradjuje se funkcija main(Meth main: none, 0, 0 ) na liniji 7
INFO  01:15:54,488 - Koriscenje  int(int: int, -1, -1 ) na liniji 8
ERROR 01:15:54,488 - Greska na liniji 8 : nije moguca dodela necemusto nije varijabla ili element niza!
ERROR 01:15:54,488 - Greska na liniji 8 : ime bla nije deklarisano! 
INFO  01:15:54,488 - Koriscenje  bla(Var noObj: none, -1, -1 ) na liniji 8
ERROR 01:15:54,488 - Greska na liniji 8 : nije moguca dodela2!
INFO  01:15:54,489 - ==================SEMANTICKA ANALIZA==============
INFO  01:15:54,489 - 1  methods in the program
INFO  01:15:54,489 - 0  global variables
INFO  01:15:54,489 - 0  global constants
INFO  01:15:54,489 - 1  global arrays
INFO  01:15:54,489 - 0  local variables in main
INFO  01:15:54,489 - 1  statements in main
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, 0, 1 
Prog test302: notype, 0, 1 
   Var niz: Arr of int, 0, 0 
   Meth main: notype, 0, 0 



ERROR 01:15:54,490 - Program nije semanticki ispravan!
